hypervisor/src/vm.rs

```diff
@@ -161,7 +162,7 @@ pub trait Vm: Send + Sync {
     /// Creates/modifies a guest physical memory slot.
     fn set_user_memory_region(&self, user_memory_region: MemoryRegion) -> Result<()>;
     /// Creates an emulated device in the kernel.
-    fn create_device(&self, device: &mut CreateDevice) -> Result<DeviceFd>;
+    fn create_device(&self, device: &mut CreateDevice) -> Result<Arc<dyn Device>>;
```

Comment:
We should also have `create_device_passthrough()` return `Result<Arc<dyn Device>>`

Reply:
For passthrough device, after creating the device, we need to call `VfioContainer::new()`, which is KVM only and requires `DeviceFd`.
If `create_passthrough_device()` returns `Result<Arc<dyn Device>>`, we will have to get the `DeviceFd` from it somehow. Is there an easy way to do that?

Reply:
We could have:

```
pub trait Device: Send + Sync + AsRawFd
```

and implement the `AsRawFd` trait from `hypervisor/src/kvm/mod.rs`
Since `kvm_ioctls::DeviceFd` implements `FromRawFd`, we should be able to move between `Device` and `DeviceFd`.

Reply:
Thanks a lot for sharing, I will try and update.

Reply:
I added a commit for `create_passthrough_device()`. But I ended up using an `unsafe` block in converting the raw_fd, which made me feel uneasy. Did I do it in wrong way?